### PR TITLE
Fix@main

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
This should fix issues relating to getting the remote name

### CODE REVIEW

File: R/git_tools.R [hunk 1]
```diff
@@ -184,15 +184,25 @@ install_external_deps <- function(repo_dir, internal_pkg_deps, ...) {
 # function to get the remote name (e.g. origin) which matches
 # the url given in the staged.deps yaml file
 get_remote_name <- function(git_repo, repo_url) {
+
+  # remove the https:// and .git from repo_url
+  repo_url <- strsplit(repo_url, "//", fixed = TRUE)[[1]][2]
+  repo_url <- gsub(".git", "", repo_url)
```
Comment (gogonzo) [inline R/git_tools.R:190]:
```suggestion
  repo_url <- gsub(".git$", "", repo_url)
```

**Review by gogonzo — COMMENTED** [review]
how do you think?
```r
git_url <- "https://github.com/x/y"
(git_url <- gsub("^.+://", "", git_url, perl = TRUE))

  


target_url <-  c(
  "https://github.com/x/y",
  "https://github.com/x/y",
  "https://github.com/x/y.git",
  "git@github.com:x/y.git",
  "git@ssh.github.com:x.y.git"
)
target_url <- gsub("^(https://|git@ssh.|git@)|\\.git$", "", target_url)
target_url <- gsub(":", "/", target_url)
target_url
# [1] "github.com/x/y" "github.com/x/y" "github.com/x/y" "github.com/x/y" "github.com/x.y"
```

File: R/git_tools.R [hunk 2]
```diff
@@ -184,15 +184,26 @@ install_external_deps <- function(repo_dir, internal_pkg_deps, ...) {
 # function to get the remote name (e.g. origin) which matches
 # the url given in the staged.deps yaml file
 get_remote_name <- function(git_repo, repo_url) {
+
+  # remove the https:// and .git from repo_url
+  repo_url <- gsub("^.+://", "", repo_url, perl = TRUE)
+  repo_url <- gsub(".git$", "", repo_url)
```
Comment (gogonzo) [inline R/git_tools.R:190]: line 190 is needed only if we assume that staged.dependencies.tyaml might contain x/y.git
If we assume so, please combine two lines into one regexp
```suggestion
  repo_url <- gsub("^.+://|.git$", "", repo_url, perl = TRUE)
```

**Review by gogonzo — APPROVED** [review]
Please combine commented regexp or remove one of them - depending on the assumption

**Review by gogonzo — APPROVED** [review]
👍 I don't think it needs a NEWS entry

**Comment by nikolas-burkoff** [comment]
Yeah I think it is covered by "Allow a remote other than origin."

**Pull request merged**